libs/ledger-live-common/src/families/cosmos/js-synchronisation.ts

```diff
@@ -54,130 +54,110 @@ const txToOps = (info: AccountShapeInfo, accountId: string, txs: CosmosTx[]): Op
 
     const correspondingMessages = messages.filter(m => m.type === mainMessage.type);
 
-    if (correspondingMessages.length === 0) {
-      continue;
-    }
-
-    // TODO: This mechanism should be removed
-    const attributes: { [id: string]: any } = {};
-    mainMessage.attributes.forEach(item => (attributes[item.key] = item.value));
-
-    // https://docs.cosmos.network/v0.42/modules/staking/07_events.html
     switch (mainMessage.type) {
       case "transfer":
-        if (attributes.sender && attributes.recipient && attributes.amount) {
-          op.senders.push(attributes.sender);
-          op.recipients.push(attributes.recipient);
-
-          if (attributes.amount.indexOf(currency.units[1].code) != -1) {
-            op.value = op.value.plus(attributes.amount.replace(currency.units[1].code, ""));
-          }
-
-          if (!op.type && attributes.sender === address) {
-            op.type = "OUT";
-            op.value = op.value.plus(fees);
-          } else if (!op.type && attributes.recipient === address) {
-            op.type = "IN";
+        // TODO: handle IBC transfers here
+        for (const message of correspondingMessages) {
+          const amount = message.attributes.find(attr => attr.key === "amount")?.value;
+          const sender = message.attributes.find(attr => attr.key === "sender")?.value;
+          const recipient = message.attributes.find(attr => attr.key === "recipient")?.value;
+          if (amount && sender && recipient && amount.indexOf(currency.units[1].code) != -1) {
+            if (op.senders.indexOf(sender) === -1) op.senders.push(sender);
+            if (op.recipients.indexOf(recipient) === -1) op.recipients.push(recipient);
+            op.value = op.value.plus(amount.replace(currency.units[1].code, ""));
+            if (!op.type && sender === address) {
```

Comment:
Why do we need to check if op type is already set ?

Reply:
yes, you are right, it is not necessary.
but I don't think it breaks anything.
anyway, I copy the same code from 
https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/families/cosmos/js-synchronisation.ts#L76

Reply:
I think we should remove useless code

Reply:
fixed. op.type check removed

Reply:
Thanks !